tests/core/executor/test_MachineTargetExecutor.py

```diff
@@ -55,6 +56,130 @@ def test_machine_executor_prepare(self):
+                register="result_create_workspace",
+                action=dict(
+                    module="file",
+                    path="{{result_pwd.stdout}}/ychaos_ws",
+                    state="directory",
+                    mode="0755",
+                ),
+            ),
+            dict(
+                name="Copy testplan from local to remote",
+                register="result_testplan_file",
+                action=dict(
+                    module="copy",
+                    content=json.dumps(
+                        mock_valid_testplan.to_serialized_dict(), indent=4
+                    ),
+                    dest="{{result_create_workspace.path}}/testplan.json",
+                ),
+            ),
+            dict(
+                name="Run YChaos Agent",
+                ignore_errors="yes",
+                action=dict(
+                    module="shell",
+                    cmd=(
+                        "source {{result_pip.virtualenv}}/bin/activate"
+                        " && "
+                        "ychaos --log-file {{result_create_workspace.path}}/ychaos.log agent attack --testplan {{result_testplan_file.dest}} --attack-report-yaml {{result_create_workspace.path}}/attack_report.yaml"
+                    ),
+                ),
+            ),
+            dict(
+                name="Zip workspace directory",
+                action=dict(
+                    module="community.general.archive",
+                    path="{{result_create_workspace.path}}",
+                    dest="{{result_create_workspace.path}}/ychaos.zip",
+                    format="zip",
+                ),
+            ),
+            dict(
+                name="Copy Workspace directory to local",
+                action=dict(
+                    module="fetch",
+                    src="{{result_create_workspace.path}}/ychaos.zip",
+                    dest=str(
+                        mock_valid_testplan.attack.get_target_config().report_dir.resolve()
+                    )
+                    + "/ychaos_{{inventory_hostname}}.zip",
+                ),
+            ),
+            dict(
+                name="Delete YChaos Workspace on host",
+                action=dict(
+                    module="file",
+                    path="{{result_create_workspace.path}}",
+                    state="absent",
+                ),
+            ),
+            dict(
+                name="Delete Virtual environment",
+                action=dict(
+                    module="file",
+                    path="{{result_pip.virtualenv}}",
+                    state="absent",
+                ),
+            ),
+        ]
+        playbook_tasks = executor.ansible_context.play_source["tasks"]
+
+        self.assertEqual(len(playbook_tasks), len(expected_tasks))
+
+        for i, task in enumerate(playbook_tasks):
+            self.assertEqual(task, expected_tasks[i])
```

Comment:
This should be Dictionary comaparison `assertDictEqual`
```suggestion
            self.assertDictEqual(task, expected_tasks[i])
```